### DISCUSSION TITLE
Having some CPU Clock speed issues while using G helper

### DISCUSSION THREAD

**Discussion by avvasai** [seerge/g-helper#3187] [Q&A]
So I recently wanted to start using G-Helper, so reset my windows and followed all the install instructions on G-hleper installation page. Recently I started noticing that my laptop when on battery, it becam unsable. So I started trying to diagnosie it and noticed that my CPU minimum clock speed would become 0.4 GHz when on battery.  So I started looking around for solutions(Things I tried):
1. Update AMD driver and Adrelain Software
2. Removed the ability of the BIOS to control armoury crate
3. Set the minimum process speed in control Panel to 100% by changing the power plan settings.
4. Tried Ryzen CPU control options for laptop CPUs.
5. Edited my Regedit to let my system use UTC clock, as some dell laptops limit CPU clock speed in dual boot configuration due to this issue.

Laptop Details:
Windows 11 and Ubuntu 18.04 Dual Boot Configuration
G14 2021 Model
![image](https://github.com/user-attachments/assets/1e295af7-7df7-4cbc-a2c9-3827400e87e2)

So I came back to this github page to see if some, solution could be found to this. Read that you use the x86 Tuning tool. Downloaded it and ran the presets. To my suprise the presets fixed my limited clockspeed issue on battery. But when I tried changing the profile from Turbo to Balanced on G-Helper to see what happens, the clock speed droppeed back to 0.4 GHz. Not sure why they are interferring, I am writing this to gain more insight. So the current solution is to use both G-helper and x86 tuning tool. Which seem counter intutitve.

Sorry, if it seems like an old mans rambling. If you need more info about this, please reach out.

Thanks.

**Comment by seerge** [comment]
@avvasai hello, G-Helper is [not a driver](https://github.com/seerge/g-helper?tab=readme-ov-file#-important-notice) and it doesn't run your hardware in realtime.  It also doesn't have any controls over CPU clocks. 

- Usually if CPU sits on it's lowest possible clocks - it's a sign that battery is not supplying enough power. Does your device run ok when plugged in ?
- Did you do any modifications to a default windows "Balanced" **power plan** , if so - try to reset it completely [to defaults as explained in Troubleshooting](https://github.com/seerge/g-helper/wiki/Troubleshooting#resetting-windows-power-plan-to-defaults)
- You can also try to [do a hard reset](https://github.com/seerge/g-helper/wiki/Troubleshooting#hard-reset--embedded-controller-ec-reset)

G-Helper uses UXTU's library only for (absolutely optional) undervolting. This is `Advanced` tab under `Fans + Power`. If you don't use that - then UXTU's library is not used as well :)

**Reply by avvasai** [reply]
Unfortunately, I have reset it to balanced power, and that still doesn't seem to work. There must be some conflicting issues that I would need to find.

**Reply by seerge** [reply]
- Try to do a hard reset as I mentioned above.
- Try to set windows "High Performance" power plan to see if it (accidentally) changes something by running following command in powershell

```
powercfg /s 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c
```

**Reply by avvasai** [reply]
I have tried a hard reset and also ran the command mentioned above. I have also tried removing the battery and discharging the CMOS battery to see if that makes a difference. I ran the commands to check the health of my Windows Image, fixed the image health, reinstalled Windows, and reverified image health. The command is some thing I ran just now, but the other things I mentioned were a part of the diagnostic processes I went through before reaching out to you.

**Reply by seerge** [reply]
Ok, how does Armoury Crate behave in same situation ?

**Reply by avvasai** [reply]
I havent installed it at all. Might reinstall to see if this issue persists. Will update in a while.
